rust/gitxetcore/src/summaries/analysis.rs

```diff
@@ -111,14 +114,18 @@ pub struct FileSummary {
     // for historical reasons this is called libmagic but does not use libmagic
     pub libmagic: Option<LibmagicSummary>,
 
+    // A buffer to allow us to add more to the serialized options
+    pub additional_summaries: Option<SummaryExt>,
+}
+
+#[derive(Serialize, Deserialize, Default, PartialEq, Clone, Debug)]
+pub struct SummaryExt {
+    pub version: u32,
```

Comment:
If the default version is 1 (given by ADDITIONAL_SUMMARY_VERSION above) then deriving Default will set this to 0.  Might just set ADDITIONAL_SUMMARY_VERSION to 0 or not derive Default?